server/types/oid/iterate.go

```diff
@@ -159,6 +179,51 @@ func IterateCurrentDatabase(ctx *sql.Context, callbacks Callbacks) error {
 	return nil
 }
 
+// iterateTypes iterates over all types that are present in the pg_type table.
+func iterateTypes(ctx *sql.Context, callbacks Callbacks) error {
+	allTypes := pgtypes.GetAllTypes()
+	// Exclude all types that are not present in the pg_type table
+	for _, t := range allTypes {
+		switch t.BaseID() {
+		case pgtypes.DoltgresTypeBaseID_Null,
+			pgtypes.DoltgresTypeBaseID_Int16Serial,
+			pgtypes.DoltgresTypeBaseID_Int32Serial,
+			pgtypes.DoltgresTypeBaseID_Int64Serial:
+			continue
+		default:
+			if _, ok := t.(pgtypes.DoltgresArrayType); !ok {
+				if _, ok = t.(pgtypes.DoltgresPolymorphicType); !ok {
+					cont, err := callbacks.Type(ctx, ItemType{
+						OID:          t.OID(),
+						StandardName: getStandardNameFromOid(t.OID()),
+						Item:         t,
+					})
+					if err != nil {
+						return err
+					}
+					if !cont {
+						return nil
+					}
+				}
+			}
+		}
+	}
+	_, err := callbacks.Type(ctx, ItemType{
+		OID:          pgtypes.InternalChar.OID(),
+		StandardName: getStandardNameFromOid(pgtypes.InternalChar.OID()),
+		Item:         pgtypes.InternalChar,
+	})
+	return err
+}
+
+// getStandardNameFromOid returns the SQL standard name of an OID if it exists.
+func getStandardNameFromOid(toid uint32) string {
```

Comment:
This is only needed for `regtype`, so we can move or remove this once we remove `StandardName` from `ItemType`.